Core/src/org/sleuthkit/autopsy/timeline/ui/listvew/datamodel/ListViewModel.java

```diff
@@ -106,11 +106,21 @@ public ListViewModel(FilteredEventsModel eventsModel) {
                 List<Long> eventIDs = unGroupConcat(resultSet.getString("eventIDs"), Long::valueOf);
                 List<EventType> eventTypes = unGroupConcat(resultSet.getString("eventTypes"),
                         typesString -> eventManager.getEventType(Integer.valueOf(typesString)).orElseThrow(() -> new TskCoreException("Error mapping event type id " + typesString + ".S")));
+                
                 Map<EventType, Long> eventMap = new HashMap<>();
-                for (int i = 0; i < eventIDs.size(); i++) {
-                    eventMap.put(eventTypes.get(i), eventIDs.get(i));
+                
+                if(hasFileTypeEvents(eventTypes)) {
+                    for (int i = 0; i < eventIDs.size(); i++) {
+                        eventMap.put(eventTypes.get(i), eventIDs.get(i));
+                    }
+                    combinedEvents.add(new CombinedEvent(resultSet.getLong("time") * 1000,   eventMap));
+                } else {
+                    for (int i = 0; i < eventIDs.size(); i++) {
+                        eventMap.put(eventTypes.get(i), eventIDs.get(i));
+                        combinedEvents.add(new CombinedEvent(resultSet.getLong("time") * 1000,   eventMap));
+                        eventMap = new HashMap<>();
```

Comment:
I'm not sure if I understand whats going on here. I understand that the  if(hasFileTypeEvents(eventTypes)) 

Why is eventMap being allocated in the loop? 

If the intent is to create a combinedEvent with a single event in eventMap each time, then a comment indicating  that would be helpful, and I'd say to move the allocation to the top of the loop.